src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfigValueSerializer.cs

```diff
@@ -28,6 +28,20 @@ private static string UnescapeLineBreaks(string str)
             parseValue: str => EscapeLineBreaks(str),
             serializeValue: UnescapeLineBreaks);
 
+        private static readonly EditorConfigValueSerializer<bool?> s_nullableBoolean = new(
+            parseValue: ParseNullableBoolean,
+            serializeValue: nullableBoolean => nullableBoolean == null ? "null" : nullableBoolean.Value.ToString());
+
+        private static Optional<bool?> ParseNullableBoolean(string str)
+        {
+            if (bool.TryParse(str, out var booleanValue))
+            {
+                return new Optional<bool?>(booleanValue);
+            }
+
+            return str == "null" ? new Optional<bool?>(null) : new Optional<bool?>();
```

Comment:
>str == "null" [](http://example.com/codeflow?start=19&length=13)

Since bool.TryParse is case-insensitive it would make sense to ignore case when comparing `null`  as well.